### ISSUE TITLE
Keycloak tries to load contents from the hostname specified in the "hostname" option instead of "hostname-admin" when accessing the Admin Console

### ISSUE THREAD

**Issue by tuxthegentoo** [keycloak/keycloak#12612]
### Describe the bug

When setting the "hostname-admin" option to a different value than "hostname", Keycloak tries to get content(s) from the URL/Hostname specified in "hostname" instead of "hostname-admin" while trying to access the Admin Console. Due to the X-Frame-Options directive being set to “SAMEORIGIN“ this behavior is blocked, resulting in a blank page.

### Version

18.0.1

### Expected behavior

Keycloak loads content(s) from the URL/Hostname specified in "hostname-admin" when accessing the Admin Console.

### Actual behavior

Keycloak tries to load content(s) from the URL/Hostname specified in "hostname" instead of "hostname-admin" when accessing the Admin Console, which is then blocked by the X-Frame-Options directive set to “SAMEORIGIN“, resulting in a blank page.

### How to Reproduce?

1. Set "hostname" to e.g. keycloak.example.com
2. Set "hostname-admin" to e.g. keycloak-admin.example.com
4. Try to access the Admin Console with keycloak-admin.example.com/admin/

The Admin Console won't load and in the Browser Console the following message can be seen:
`The loading of “https://keycloak.example.com/realms/master/protocol/openid-connect/3p-cookies/step1.html?version=19iuw” in a frame is denied by “X-Frame-Options“ directive set to “SAMEORIGIN“.`

### Anything else?

Options I am using to start/build Keycloak:
`--proxy=edge --cache=ispn --http-enabled=true --hostname=keycloak.example.com --hostname-admin=keycloak-admin.example.com --hostname-strict=true --hostname-strict-backchannel=true`

Issue seems to be related to #10340.

**Comment by stianst** [comment]
Problem here is the X-Frame-Options as those resources should be loaded from keycloak.example.com as they are part of the OIDC session management endpoint, and not part of the admin endpoints.

**Comment by pedroigor** [comment]
@tuxthegentoo I can't reproduce the issue. I'm starting the server using the same options but changing the host names:

```
./kc.sh start --auto-build --cache local --hostname mykeycloak --hostname-admin mykeycloakadmin --proxy edge --health-enabled true
```

In front of the server, there is an HAProxy running.

Would you be able to give more details about your environment setup and help me to reproduce the problem locally?

**Comment by pedroigor** [comment]
@tuxthegentoo Thanks. That is what I was wondering and the reason about asking you the config.

Glad it worked and the "security headers" can be managed from the realm settings too. By default, that one is included.

**Comment by mazenaissa** [comment]
Hello guys,
I have the same issue although I'm not setting add_header X-Frame-Options "SAMEORIGIN"; in my ingress-inginx-controller.
I believe the request was denied due to the header X-Frame-Options "SAMEORIGIN"  from the ream master settings. I am able to resolve the issue and connect to the console after deleting KC_HOSTNAME and adding KC_HOSTNAME_STRICT=false, but I don't this variable to be false for security reasons. Any idea how to solve it?
Thanks,

**Comment by mazenaissa** [comment]
> Hello guys, I have the same issue although I'm not setting add_header X-Frame-Options "SAMEORIGIN"; in my ingress-inginx-controller. I believe the request was denied due to the header X-Frame-Options "SAMEORIGIN" from the ream master settings. I am able to resolve the issue and connect to the console after deleting KC_HOSTNAME and adding KC_HOSTNAME_STRICT=false, but I don't want this variable to be false for security reasons. Any idea how to solve it? Thanks,

Well after further investigation it seems that the request was denied because the KC_HOSTNAME is the one who sent the X-Frame-Options "SAMEORIGIN" and the header config in the realm is not related to this issue.

**Comment by mazenaissa** [comment]
@pedroigor How can I fix the issue of the iframe without removing X-Frame-Options "SAMEORIGIN" or deleting KC_HOSTNAME and adding KC_HOSTNAME_STRICT=false ?

**Comment by mazenaissa** [comment]
@pedroigor I'll be grateful if you helped me here https://github.com/keycloak/keycloak/issues/12695
